sources/packages/web/src/views/institution/locations/programs/LocationProgramOffering.vue

```diff
@@ -27,26 +29,59 @@ export default {
       type: Number,
       required: true,
     },
+    offeringId: {
+      type: Number,
+      required: true,
+    },
   },
   setup(props: any) {
     const toast = useToast();
     const router = useRouter();
-    const submitted = async (data: any) => {
-      try {
-        await EducationProgramOfferingService.shared.createProgramOffering(
+    const initialData = ref({});
+    onMounted(async () => {
+      if (props.offeringId) {
+        initialData.value = await EducationProgramOfferingService.shared.getProgramOffering(
           props.locationId,
           props.programId,
-          data,
+          props.offeringId,
         );
+      }
+    });
+
+    const submitted = async (data: any) => {
+      try {
+        if (props.offeringId) {
+          await EducationProgramOfferingService.shared.updateProgramOffering(
+            props.locationId,
+            props.programId,
+            props.offeringId,
+            data,
+          );
+        } else {
+          await EducationProgramOfferingService.shared.createProgramOffering(
+            props.locationId,
+            props.programId,
+            data,
+          );
+        }
         router.push({
           name: InstitutionRoutesConst.VIEW_LOCATION_PROGRAMS,
         });
-        toast.add({
-          severity: "success",
-          summary: "Created!",
-          detail: "Education Offering created successfully!",
-          life: 5000,
-        });
+        if (props.offeringId) {
```

Comment:
The logic to display the toast notification could be moved inside the previous "if (props.offeringId)" after line 59 and 65 to avoid checking the same condition again.
Or the toast notification could be simplified to something generic for create/edit like "Education Offering saved successfully!"